File: DS/python/stack.py
class stack:
    def __init__(self):
        self.stack = []

    def push(self, obj):
        self.stack.append(obj)

    def pull(self):
        if len(self.stack) > 0:
            return self.stack.pop()
        return None

    def top(self):
        if len(self.stack) > 0:
            return self.stack[-1]
        return None

File: DS/C/stack.c
#include <stdio.h>

typedef struct _stack{
	char value[100];
	int top;
} _stack;

void PUSH(char c, _stack * stack){
	stack->top++;
	stack->value[stack->top] = c;
}

char POP(_stack * stack){
	char c = stack->value[stack->top];
	stack->top--;
	return c;
}
